### PR TITLE
fix: [DHIS2-10702] - Unexpected error querying tracked entity instance porting [2.36]

### PR DESCRIPTION
**Pull request by lucaCambi77**
see https://github.com/dhis2/dhis2-core/pull/7722

### CODE REVIEW

**Review by gnespolino — APPROVED** [review]
Please check license, it should be 2021 everywhere as mentioned in the other PR

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityTypeObjectBundleHook.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import org.hisp.dhis.preheat.PreheatIdentifier;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.hisp.dhis.trackedentity.TrackedEntityType;
+import org.springframework.stereotype.Component;
+
+@Component( "org.hisp.dhis.dxf2.metadata.objectbundle.hooks.TrackedEntityTypeObjectBundleHook" )
+public class TrackedEntityTypeObjectBundleHook extends AbstractObjectBundleHook
+{
+    @Override
+    public <T extends IdentifiableObject> List<ErrorReport> validate( T object, ObjectBundle bundle )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityTypeObjectBundleHook.java:75]: Maybe add Javadoc to explain what is going on there. Streams/lambdas are not that easy to read :-)

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.33/V2_33_36__Add_translations_column_into_programruleaction_table.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+alter table programruleaction add column if not exists translations jsonb;
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.33/V2_33_36__Add_translations_column_into_programruleaction_table.sql:1]: Is this file intentional? Does not seem related?

Comment (lucaCambi77) [reply dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.33/V2_33_36__Add_translations_column_into_programruleaction_table.sql:1]: this is a script that it has been renamed as it had a wrong name so when adding my script i had to rename it for proper progression

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityTypeObjectBundleHook.java (hunk 1 above)
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityTypeObjectBundleHook.java:75]: i am not a supporter of java doc as names should explains but yes we can add some comments

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.33/V2_33_36__Add_translations_column_into_programruleaction_table.sql (hunk 2 above)
Comment (stian-sandvold) [reply dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.33/V2_33_36__Add_translations_column_into_programruleaction_table.sql:1]: This is a script for 2.33, and not 2.36, so it should be fixed on the 2.33 branch, not 2.36. Or is this a relevant problem for 2.36?

I would in either case suggest removing this file from this PR, and fixing it in a separate PR

**Review by stian-sandvold — CHANGES_REQUESTED** [review]
See comment about flywayscript for 2.33

**Review by stian-sandvold — APPROVED** [review]
